File: 201-Bitwise AND of Numbers Range.py
class Solution(object):
    def rangeBitwiseAnd(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: int
        """
# bit shift
        shift = 0
        while left < right:
            left >>= 1
            right >>= 1
            shift += 1
            
        return right << shift
    
# Brian Kernighan's Algorithm
        while right!=0 and right>left:
            right = right & (right-1)
            
        return right

File: JavaScript/201-Bitwise AND of Numbers Range.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeBitwiseAnd = function(left, right) {
    while (right !== 0 && right > left){
        right = right&(right-1);
    }
    return right;
};